From bd876d67b0ea6fa4cac6c371219021521cd7c5d3 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Tue, 25 Oct 2022 16:28:26 +0300
Subject: [PATCH] [SW-71413] scripts: gaudi3: add elfs tgz

Add a way to archive zephyr elf files for debug purposes.

Usage (after build):
make gaudi3-debug-elfs-create

Change-Id: I9d24fbba7e7d1abfe48ecdb8ce78f1dfd98b17e7
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/scripts/make/gaudi3/Makefile b/scripts/make/gaudi3/Makefile
index a17dfc5..0e80f3a 100644
--- a/scripts/make/gaudi3/Makefile
+++ b/scripts/make/gaudi3/Makefile
@@ -1,2 +1,14 @@
 .PHONY: gaudi3-post-image
 gaudi3-post-image:
+
+.PHONY: gaudi3-debug-elfs-create
+gaudi3-debug-elfs-create:
+	@mkdir -p $(IMAGES_DIR)/elfs
+	@for elf in $$(ls $(BR2_EXTERNAL)/output/gaudi3_arc/build/zephyr-custom/*/zephyr/zephyr.elf); do \
+		cp $${elf} $(IMAGES_DIR)/elfs/`echo $${elf} | sed "s@.*/\(.*\)/zephyr/zephyr.elf@\1.elf@"`; \
+	done
+	@tar czf $(IMAGES_DIR)/${HL_DEV}-elfs.tgz -C $(IMAGES_DIR)/ elfs
+	@echo
+	@echo Created $(IMAGES_DIR)/${HL_DEV}-elfs.tgz
+	@echo
+
